Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -922,21 +922,31 @@ namespace AzToolsFramework
         return false;
     }
 
-    void ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
+    ComponentEditor::ComponentCardState ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
     {
+        if (componentModeTypes.empty())
+        {
+            SetWidgetInteractEnabled(this, false);
+            return ComponentCardState::Disabled;
+        }
+
         // disable all component cards not matching the ComponentMode type
-        if (AZStd::find(
-            componentModeTypes.begin(),
-            componentModeTypes.end(), m_componentType) == componentModeTypes.end())
+        if (AZStd::find(componentModeTypes.begin(), componentModeTypes.end(), m_componentType) == componentModeTypes.end())
         {
             SetWidgetInteractEnabled(this, false);
+            return ComponentCardState::Disabled;
         }
         else
         {
-            if (!componentModeTypes.empty())
+            if (componentModeTypes.front() == m_componentType)
             {
                 // only set the first item to be selected/highlighted
-                SetSelected(componentModeTypes.front() == m_componentType);
+                SetSelected(true);
+                return ComponentCardState::Selected;
+            }
```

Comment:
I apologize for the conflicting guidance here, but to reiterate from my previous comment, there's no guarantee that SetSelected() is actually selecting the item:
```
    void ComponentEditor::SetSelected(bool selected)
    {
        // do not allow cards to be selected when they are disabled
        // (show the yellow outline)
        if (!isEnabled())
        {
            return;
        }

        AzQtComponents::Card::setSelected(selected);
    }
```
I realize that to get to this point where the component is on the list of componentModeTypes, it should very _likely_ always be selectable based on the current editor logic, I'm just suggesting that this is introducing a point of fragility should any logic elsewhere in the Editor ever change. From that point of view, is there any reason to use this enum at all, vs just calling IsSelected() in the places that are adjusting the scrolling?

Reply:
This is a fair point and I discussed this with @amzn-ahmadkrm but I think for this change this is outside the scope of the PR as `SetSelected` already doesn't return if it succeeded or not and we were already relying on this logic before so I think leaving as-is for now is okay and we can always revisit this in future if other problems arise.